plugins/auth/auth_base.js

```diff
@@ -101,20 +101,28 @@ exports.check_user = function (next, connection, credentials, method) {
         return;
     }
 
-    const passwd_ok = function (valid, message) {
+    // valid: (true|false)
+    // opts: ({ message, code }|String)
+    var passwd_ok = function (valid, opts) {
+        var status_code = (typeof(opts) == 'object' && opts['code']) || (valid ? 235 : 535);
+        var status_message = (typeof(opts) == 'object' ? opts['message'] : opts) ||
+                (valid  ? '2.7.0 Authentication successful' : '5.7.8 Authentication failed');
+
         if (valid) {
             connection.relaying = true;
             connection.results.add({name:'relay'}, {pass: plugin.name});
+            connection.results.add(plugin, {pass: method});
+
             connection.results.add({name:'auth'}, {
                 pass: plugin.name,
                 method: method,
-                user: credentials[0],
+                user: credentials[0]
```

Comment:
You don't need to agree that trailing commas in lists are preferable, and contributions aren't required to have them, but also please don't remove them.